From cc80d61acebcacdc0d68ac566c4ae4bf35f7a014 Mon Sep 17 00:00:00 2001
From: Matt Porter <mporter@ti.com>
Date: Thu, 20 Sep 2012 00:36:44 -0400
Subject: [PATCH 17/17] ARM: dts: add am33xx EDMA support

Adds support for the defined EDMA, generic DMA controller, and
DMA request bindings for mmc and spi.

Signed-off-by: Matt Porter <mporter@ti.com>
---
 arch/arm/boot/dts/am33xx.dtsi |   50 +++++++++++++++++++++++++++++++++++++++++
 1 files changed, 50 insertions(+), 0 deletions(-)

diff --git a/arch/arm/boot/dts/am33xx.dtsi b/arch/arm/boot/dts/am33xx.dtsi
index bc864ab..edf220e 100644
--- a/arch/arm/boot/dts/am33xx.dtsi
+++ b/arch/arm/boot/dts/am33xx.dtsi
@@ -86,6 +86,36 @@
 			reg = <0x48200000 0x1000>;
 		};
 
+		edma: edma@49000000 {
+			compatible = "ti,edma3";
+			ti,hwmods = "tpcc", "tptc0", "tptc1", "tptc2";
+			reg = <0x49000000 0x10000>;
+			interrupt-parent = <&intc>;
+			interrupts = <12 13 14>;
+			#dma-cells = <1>;
+			dma-channels = <64>;
+			ti,edma-regions = <4>;
+			ti,edma-slots = <256>;
+			ti,edma-reserved-channels = <0  2
+					     14 2
+					     26 6
+					     48 4
+					     56 8>;
+			ti,edma-reserved-slots = <0  2
+					  14 2
+					  26 6
+					  48 4
+					  56 8
+					  64 127>;
+			ti,edma-queue-tc-map = <0 0
+					1 1
+					2 2>;
+			ti,edma-queue-priority-map = <0 0
+					      1 1
+					      2 2>;
+			ti,edma-default-queue = <0>;
+		};
+
 		gpio0: gpio@44e07000 {
 			compatible = "ti,omap4-gpio";
 			ti,hwmods = "gpio1";
@@ -288,11 +318,21 @@
 		mmc1: mmc@48060000 {
 			compatible = "ti,omap3-hsmmc";
 			ti,hwmods = "mmc1";
+			ti,dual-volt;
+			ti,needs-special-reset;
+			dmas = <&edma 24
+				&edma 25>;
+			dma-names = "tx", "rx";
 		};
 
 		mmc2: mmc@481D8000 {
 			compatible = "ti,omap3-hsmmc";
 			ti,hwmods = "mmc2";
+			ti,needs-special-reset;
+			status = "disabled";
+			dmas = <&edma 2
+				&edma 3>;
+			dma-names = "tx", "rx";
 		};
 
 		mmc3: mmc@47810000 {
@@ -333,6 +373,11 @@
 			reg = <0x4830000 0x400>;
 			interrupt-parent = <&intc>;
 			interrupt = <65>;
+			dmas = <&edma 16
+				&edma 17
+				&edma 18
+				&edma 19>;
+			dma-names = "tx0", "rx0", "tx1", "rx1";
 			ti,spi-num-cs = <2>;
 			ti,hwmods = "spi0";
 			status = "disabled";
@@ -346,6 +391,11 @@
 			reg = <0x481a0000 0x400>;
 			interrupt-parent = <&intc>;
 			interrupt = <125>;
+			dmas = <&edma 42
+				&edma 43
+				&edma 44
+				&edma 45>;
+			dma-names = "tx0", "rx0", "tx1", "rx1";
 			ti,spi-num-cs = <2>;
 			ti,hwmods = "spi1";
 			status = "disabled";
-- 
1.7.7.6

